core/src/main/java/com/linecorp/armeria/client/Http1ResponseDecoder.java

```diff
@@ -161,8 +161,13 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                             state = State.NEED_DATA_OR_TRAILERS;
                         }
 
-                        res.scheduleTimeout(channel().eventLoop());
-                        res.write(ArmeriaHttpUtil.toArmeria(nettyRes));
+                        // If this tryWrite() returns false, it means the response stream has been closed due to
+                        // disconnection or by the response consumer. We do not need to handle such cases here
+                        // because it will be notified to the response consumer anyway.
+                        if (!res.tryWrite(ArmeriaHttpUtil.toArmeria(nettyRes))) {
+                            // Schedule only when the response stream is still open.
+                            res.scheduleTimeout(ctx.channel().eventLoop());
```

Comment:
Wouldn't this make timeout schedule a little bit late, because `write()` may invoke the subscriber directly?

Reply:
Yes, I can fix it. Thank!
I brought this logic from `Http2ResponseDecoder` and I thought it's fine unless the subscriber blocks the `EventLoop`.
But in `scheduleTimeout` method, it checks `isOpen()`, so we can call `scheduleTimeout` first. :-)

Reply:
Thanks! Approved.